### PR TITLE
Cleanup vagrant_provision.sh

### PR DESCRIPTION
**Pull request by willkoh**
Removes unnessary redundant downloads to speed up re-provisioning.
- Gtest, Gmock, Valgrind, and Protobuf are checked to see if they're already existing. The valgrind and protobuf installations take a long time, so not doing it again speeds up re-provisioning.

Checks to see if an existing tar file is already installed and deletes it before downloading again.
- Turns out that if provisioning fails due to different reasons (spotty internet connection being one of them), then the corrupted tar files that were downloaded are kept, and newer downloads are named as something.tar.gz1, something.tar.gz2, etc. This causes provisioning to fail every single time after such cases since it tries to un-tar the corrupted tar file instead of the newly download ones. Therefore, I changed it so that all those previous tar files are deleted before attempting to download new ones.

Fixes $PATH so that conversion between hadoop2 and hadoop3 is easier.
- Previously, hadoop2 was set in path, which makes conversion between hadoop2 and hadoop3 difficult (had to remove hadoop2 from PATH variable using `export PATH=${PATH//:\/home\/vagrant\/hadoop2\/bin:/:}`) But since /home/vagrant/hadoop/bin is in PATH, then we can remove that unnecessary PATH variable and simply use the symlink. For example, right now by default, we have hadoop2 linked to hadoop (ln -s hadoop2 hadoop in line 72). If we wanted to switch to using hadoop3, we can simply go to /home/vagrant, call `rm hadoop` and do `ln -s hadoop3 hadoop` and it'll change. Same goes for going from hadoop3 to hadoop2. This change doesn't change anything that others will be working with. It simply makes it easier for storage team to convert from hadoop2 to hadoop3.

### CODE REVIEW

File: vagrant_provision.sh [hunk 1]
```diff
@@ -47,15 +52,19 @@ cat /home/vagrant/rdfs/config/hdfs-site.xml > /home/vagrant/hadoop3/etc/hadoop/h
 cat /home/vagrant/rdfs/config/core-site.xml > /home/vagrant/hadoop3/etc/hadoop/core-site.xml
 
 # add hadoop to path
-echo 'export PATH=/home/vagrant/hadoop2/bin:$PATH' >> /home/vagrant/.bashrc
+echo 'export PATH=${PATH//:\/home\/vagrant\/hadoop2\/bin:/:}' >> /home/vagrant/.bashrc
```
Comment (LINKIWI) [inline vagrant_provision.sh:55]: If the desired version of hadoop is symlinked to `$HOME/hadoop`, what is the purpose of this line?

Comment (willkoh) [reply vagrant_provision.sh:55]: So, if you don’t destroy and up, then the path will still have hadoop2. That line is to get rid of it for people who are just re-provisioning.

Comment (LINKIWI) [reply vagrant_provision.sh:55]: Cool makes sense

**Review by LINKIWI — APPROVED** [review]
Thanks

**Pull request merged**